guides/hack/15-asynchronous-operations/19-extensions.md

```diff
@@ -3,24 +3,64 @@ infrastructure. Async is especially useful with database access and caching, web
 
 ## MySQL
 
-The async MySQL extension is similar to the [`mysqli`](http://php.net/manual/en/book.mysqli.php) extension that comes with HHVM. The async
-MySQL extension is primarily used for asynchronously creating connections and querying MySQL databases.
+The async MySQL extension is a built-in way to execute queries against databases while running Hack code when the database is processing.
+The async MySQL extension is primarily used for asynchronously creating connections and querying MySQL databases.
 
 The [full API](../reference/class/AsyncMysqlConnection/) contains all of the classes and methods available for accessing MySQL via async; we
 will only cover a few of the more common scenarios here.
 
 The primary class for connecting to a MySQL database is [`AsyncMysqlConnectionPool`](../reference/class/AsyncMysqlClient/) and its
 primary method is the `async` [`connect`](../reference/class/AsyncMysqlClient/connect/).
 
-The primary class for querying a database is [`AsyncMysqlConnection`](../reference/class/AsyncMysqlConnection/) with the two main
-query methods, [`query`](../reference/class/AsyncMysqlConnection/query/) and [`queryf`](../reference/class/AsyncMysqlConnection/queryf/),
-both of which are `async`. There is also a function to ensure that queries to be executed are safe called
+The primary class for querying a database is [`AsyncMysqlConnection`](../reference/class/AsyncMysqlConnection/) with the three main query methods:
+[`query`](../reference/class/AsyncMysqlConnection/query/),
+[`queryf`](../reference/class/AsyncMysqlConnection/queryf/), and
+[`queryAsync`](../references/class/AsyncMysqlConnection/queryAsync)
+all of which are `async`. The naming conventions for async method have not been applied consistently.
+There is also a function to ensure that queries to be executed are safe called
 [`escapeString`](../reference/class/AsyncMysqlConnection/escapeString/).
 
+## Choosing a query method
+
+Choosing the right tool for the job is very important.
+Each of the three methods has its strengths and weaknesses.
+
+### [`queryf`](../reference/class/AsyncMysqlConnection/queryf/)
+This should be your preferred method. It is powerful enough to express all queries with a known structure, but with unknown data.
```

Comment:
queryf should be deprecrated; `queryAsync()` should pretty much always be preferred:
- it uses hack arrays instead of legacy hack collections
- it matches current naming conventions

Reply:
It also makes it possible to pass queries and data independent of connections, which in turn allows building much more powerful abstractions on top of it